pkg/commands/isrunning/isrunning.go

```diff
@@ -0,0 +1,85 @@
+
+import (
+	"fmt"
+
+	"github.com/openshift/configuration-anomaly-detection/pkg/aws"
+	"github.com/openshift/configuration-anomaly-detection/pkg/ocm"
+
+	"github.com/aws/aws-sdk-go/service/cloudtrail"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	v1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+)
+
+// IsRunningImpl should have all of the functions that IsRunningService is implementing
+type IsRunningImpl struct {
+	AwsClient aws.Client
+	OcmClient ocm.Client
+}
+
+// IsRunningService will wrap all the required commands IsRunning needs to run it's operations
+//go:generate mockgen -source $GOFILE -destination ./mock/interfaces.go -package mock
+type IsRunningService interface {
+	// AWS
+	ListStopInstancesEvents() ([]*cloudtrail.Event, error)
+	ListStoppedInstances(infraId string) ([]*ec2.Instance, error)
+	// OCM
+	GetClusterDeployment(clusterID string) (*hivev1.ClusterDeployment, error)
+	GetClusterInfo(identifier string) (*v1.Cluster, error)
+}
+
+// IsRunning is an implementation of the Interface, and adds functionality above it
+// to craete IsRunning you can use a mock, or fill it with IsRunningImpl
+type IsRunning struct {
+	IsRunningService
+}
+
+// isUserAllowedToStop will verify if a user is allowed to stop instances.
+// as this is the thing that might change the most it is at the top.
+// additionally it is private as I don't see anyone using this outside of AreAllInstancesRunning
+func isUserAllowedToStop(username string) bool {
+	return false
+}
+
+// AreAllInstancesRunning will check all of the instances under the externalID are running and return false if any of them are not
+func (i IsRunning) AreAllInstancesRunning(externalID string) (bool, error) {
+	cluster, err := i.GetClusterInfo(externalID)
+	if err != nil {
+		return false, fmt.Errorf("could not retrieve cluster info for %s: %w", externalID, err)
+	}
+	id := cluster.ID()
+
+	cd, err := i.GetClusterDeployment(id)
+	if err != nil {
+		return false, fmt.Errorf("could not retrieve Cluster Deployment for %s: %w", id, err)
+	}
+	infraID := cd.Spec.ClusterMetadata.InfraID
+
+	stoppedInstances, err := i.ListStoppedInstances(infraID)
+	if err != nil {
+		return false, fmt.Errorf("could not retrieve instances for %s: %w", infraID, err)
+	}
+
+	// TODO: this statement is not fully what I wanted, but makes the code liner. will come back to it later
+	if len(stoppedInstances) == 0 {
+		return true, nil
+	}
+
+	stoppedInstancesEvents, err := i.ListStopInstancesEvents()
+	if err != nil {
+		return false, fmt.Errorf("could not retrieve stopped instance event: %w", err)
+	}
+
+	if len(stoppedInstancesEvents) == 0 {
+		return false, fmt.Errorf("the are stopped instances but no stoppedInstancesEvents, this means the instances were stopped too long ago or CloudTrail is not up to date")
+	}
+
+	for _, event := range stoppedInstancesEvents {
+		if !isUserAllowedToStop(*event.Username) {
+			return false, nil
+		}
+	}
```

Comment:
this method is very clean but doesn't do some simple things:
- check if multiple stop events have happened to a similar instance
- check if some stopped instances don't have a correlating event
- what about the amount of instances in the cluster + the amount of running

Reply:
solved point 3, but the others are not addressed

Reply:
resolved these in the aws package